analyzer/callable-index/src/test/java/eu/fasten/analyzer/callableindex/CallableIndexServerPluginTest.java

```diff
@@ -44,37 +48,35 @@ public void setUp() {
     @Test
     public void saveToDatabaseTest() throws IOException, RocksDBException {
         var rocksDao = Mockito.mock(RocksDao.class);
-        var json = new JSONObject("{\"payload\": {" +
-                "\"index\": 1," +
-                "\"product\": \"test\"," +
-                "\"version\": \"0.0.1\"," +
-                "\"nodes\": [1, 2, 3]," +
-                "\"numInternalNodes\": 2," +
-                "\"edges\": [[1, 2], [2, 3]]" +
-                "}}");
-        var graph = GidGraph.getGraph(json.getJSONObject("payload"));
+        var json = new JSONObject("{\"payload\": {}}");
+
+        var getTestResource = Objects.requireNonNull(Thread.currentThread().getContextClassLoader().getResource("gid_graph_test.json"));
+        json.getJSONObject("payload").put("dir", getTestResource.getPath());
```

Comment:
I am not a big fan of storing the JSON on disk versus declaring it in the class, but I acknowledge that there are benefits/drawbacks for both solutions and I do not have a strong preference... in this concrete case, however, we need to change contents (add the temp dir), so I would argue that keeping everything in one place makes it easier to understand here. I would suggest to revert back to the old declaration as string. What do you folks think?

Reply:
Since the plug-in now reads GIDs from the FS, it expects the JSON input to have a path rather than the whole graph. If you know a trick to pass the GID graph as a string for testing let me know.

Reply:
Ok, let's postpone this discussion for now to the point when we focus our refactoring efforts to the CallableIndexer